stemcell_builder/stages/bosh_go_agent/assets/restrict-nats-api-access-allow-monit-api

```
@@ -0,0 +1,57 @@
+#!/bin/bash
+# THESE SCRIPTS ARE EXECUTED IN LEXICAL ORDER. MAKE SURE TO CHECK IF ANYTHING ELSE (CURRENTLY MONIT) WRITES OTHER CGROUP BASED RULES TO AVOID INTERFERENCE. IT'S PROBABLY AGOOD IDEA TO -A DROPS AND -I ACCEPTS
+
+source /var/vcap/bosh/etc/nats-access-helper.sh
+
+
+# Give the agent access to Monit right away
+if iptables -t mangle -C POSTROUTING -d 127.0.0.1 -p tcp --dport 2822 \
+            -m cgroup --cgroup "${nats_isolation_classid}" -j ACCEPT
+then
+  /bin/true
+else
+    iptables -t mangle -I POSTROUTING -d 127.0.0.1 -p tcp --dport 2822 \
+             -m cgroup --cgroup "${nats_isolation_classid}" -j ACCEPT
+fi
+
+wait_for_agent_settings() {
+  until [[ -f /var/vcap/bosh/settings.json ]]; do
+    echo 'waiting for agent_settings.json'; 
+    sleep 10; 
+  done
+
+  NATS_URL=$(cat /var/vcap/bosh/settings.json | sed 's#.*\("mbus"\):"\(.*\)",.*#\2#g' )
+
+  # CHECK IF THIS IS A CREATE ENV.
+  if grep '^https' <<< ${NATS_URL}; then
+    echo "create-env, skipping iptables for nats"
+  else
+
+    BOSH_IP_PORT=$( cut -f2 -d@ <<< ${NATS_URL} )
+    BOSH_IP=$( cut -f1 -d: <<< ${BOSH_IP_PORT} )
+    BOSH_PORT=$( cut -f2 -d: <<< ${BOSH_IP_PORT} )
+
+    if [[ "${BOSH_IP_PORT}" != "" ]]; then
+      if iptables -t mangle -C POSTROUTING -d ${BOSH_IP} -p tcp --dport ${BOSH_PORT} \
+            -m cgroup --cgroup "${nats_isolation_classid}" -j ACCEPT
```

If you quote `"${nats_isolation_classid}"` then I suggest you quote everything like `${BOSH_IP}` and others.

I mean if `BOSH_IP` is not supposed to contain spaces but does by mistake, then you want to be warned about it and get some crash that tells you. (the “let it crash” principle)

---

https://github.com/cloudfoundry/bosh-linux-stemcell-builder/pull/197/files#diff-dcbc701279c71464bd174518868bb5a356df838db687bcd65c3867b12bee91a4R32-R43